code/game/objects/structures/handrail.dm

```diff
@@ -0,0 +1,61 @@
+/obj/structure/barricade/handrail
+	name = "handrail"
+	desc = "A railing, for your hands. Woooow."
+	icon = 'icons/obj/structures/handrail.dmi'
+	icon_state = "handrail_a_0"
+	barricade_type = "handrail"
+	stack_type = /obj/item/stack/sheet/metal
+	stack_amount = 2
+	destroyed_stack_amount = 1
+	can_wire = FALSE
+	coverage = 10
+	can_change_dmg_state = FALSE
+	resistance_flags = XENO_DAMAGEABLE
+
+/obj/structure/barricade/handrail/update_icon_state()
+	. = ..()
+	switch(dir)
+		if(SOUTH)
+			layer = ABOVE_MOB_LAYER
+		if(NORTH)
+			layer = initial(layer) - 0.01
+		else
+			layer = initial(layer)
+	if(!anchored)
+		layer = initial(layer)
```

Comment:
on this is done (incorrectly) in a lot of places including by myself in the past, but there's actuall a proc setDir which should be used for handling any changes from a dir change. If all of cades needs this though I can do it in a separate pr or something.

Reply:
Is this actually needed? After the handrail is set it's impossible to rotate again, which is why I suspect we're checking dir in Init.